paasta_tools/config_utils.py

```diff
@@ -259,3 +261,82 @@ def commit_to_remote(self, extra_message: str = ""):
             _push_to_remote(self.branch)
         else:
             log.info("No files changed, no push required.")
+
+    def merge_recommendations(
+        self, recs: Dict[Tuple[str, str], Dict[str, Any]]
+    ) -> List[Dict[Tuple[str, str], Dict[str, Any]]]:
```

Comment:
might be nice to have a docstring explaining what all these `str`s are.

```suggestion
    ) -> List[Dict[Tuple[str, str], Dict[str, Any]]]:
    """
    :param recs: Dictionary of (service, instance_type_cluster) -> recommendations. instance_type_cluster is e.g. "kubernetes-pnw-prod".
    :returns: a dictionary of the same format, with the previous recommendations merged in and autotune_limits applied.
    """
```

Reply:
can do!

Reply:
maybe tangential question but is there a technical reason we are going from a Dict of (service, instance_type_cluster) -> recommendations to a list of individual one-item {(service, instance_type_cluster): recs} dicts?

Reply:
@jfongatyelp not really - it just fit my mental model better while I was writing this (and make functions as pure as possible)

I can try to keep the stricture the same if we'd like

Reply:
ah thanks, i was just asking for my own education. Seems fine to leave, the typing makes this difference easy to see anyhow!

Reply:
i've refactored this a bit since upon further inspection i just needed to think a tiny bit harder to make this work with the same types :p